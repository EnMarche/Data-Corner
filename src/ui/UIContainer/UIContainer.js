import { Grid } from '@mui/material'
import makeStyles from '@mui/styles/makeStyles'
import PropTypes from 'prop-types'

const useStyles = makeStyles(theme => ({
  root: {
    background: theme.palette.whiteCorner,
    borderRadius: '6px',
    boxShadow: '0 1px 1px 0 rgba(0, 0, 0, .04)',
    textAlign: textAlign => textAlign.textAlign,
  },
}))

const UIContainer = ({
  children,
  rootClasses = '',
  breakpoints = { xs: null, sm: null, md: null, lg: null, xl: null },
  textAlign = null,
}) => {
  const classes = useStyles({ textAlign })

  return (
    <Grid
      item
      className={`${classes.root} ${rootClasses}`}
      xs={breakpoints.xs}
      sm={breakpoints.sm}
      md={breakpoints.md}
      lg={breakpoints.lg}
      xl={breakpoints.xl}
    >
      {children}
    </Grid>
  )
}

export default UIContainer

UIContainer.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.object]).isRequired,
  rootClasses: PropTypes.string,
  textAlign: PropTypes.string,
  breakpoints: PropTypes.shape({
    xs: PropTypes.number,
    sm: PropTypes.number,
    md: PropTypes.number,
    lg: PropTypes.number,
    xl: PropTypes.number,
  }),
}
