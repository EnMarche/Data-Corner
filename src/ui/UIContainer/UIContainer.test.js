import { render } from '@testing-library/react'
import UIContainer from './UIContainer'

jest.mock('@mui/material', () => ({
  Grid: ({ item, children, ...props }) => <div {...props}>{children}</div>,
  makeStyles: () => () => ({ root: 'root' }),
}))

describe('UIContainer', () => {
  it('displays grid', () => {
    const { container } = render(
      <UIContainer rootClasses="fooClass" xs={0} sm={1} md={2} lg={3} xl={4}>
        Foo
      </UIContainer>
    )

    expect(container).toMatchSnapshot()
  })
})
