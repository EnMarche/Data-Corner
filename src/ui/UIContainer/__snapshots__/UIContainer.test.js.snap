// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIContainer displays grid 1`] = `
<div>
  <div
    class="root fooClass"
  >
    Foo
  </div>
</div>
`;
